app/controllers/graphic_requests_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+class GraphicRequestsController < ApplicationController
+
+  def show
+    graphic_request = GraphicRequest.where(download_hash: params[:id]).first
```

Comment:
true, will use find by the download_hash so that the ApplicationController can handle the exception.